### PR TITLE
Bureaucracy module. Iteration 2. Hiring flow migration.

### PR DESCRIPTION
**Pull request by shamil-gadelshin**
### Features
- this PR contains migrated "hiring flow" extrinsics from the content working group module
- tests for migrated extrinsics
- fixed tests for the forum module
- the 'curator' entity was renamed to the 'worker'

### Extrinsics
+ add_worker_opening
+ accept_worker_applications
+ begin_worker_applicant_review
+ fill_worker_opening
+ withdraw_worker_application
+ terminate_worker_application
+ apply_on_worker_opening

#### Open question
**How should we create a mint for the bureaucracy instance?** 

In the content working group the mint is created during the genesis config initialization:
```
add_extra_genesis {
        config(mint_capacity): minting::BalanceOf<T>;
        // config(mint_adjustment): minting::Adjustment<BalanceOf<T>, T::BlockNumber> (add serialize/deserialize derivation for type)
        build(|config: &GenesisConfig<T>| {
            // create mint
            let mint_id = <minting::Module<T>>::add_mint(config.mint_capacity, None).expect("Failed to create a mint for the content working group");
            Mint::<T>::put(mint_id);
        });
    }
```

**Should we add a) the same config initialization or b) create a migration or c) an extrinsic?**

### CODE REVIEW

File: runtime-modules/bureaucracy/src/constraints.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+use codec::{Decode, Encode};
```
Comment (bedeho) [inline runtime-modules/bureaucracy/src/constraints.rs]: Can we instead use a version of this from the common module that you added?

**Review by bedeho — CHANGES_REQUESTED** [review]
Pretty much perfect, nice job!

**Comment by bedeho** [comment]
For the mint, let's wait until we get to the issue of when we want to deploy this.

File: runtime-modules/bureaucracy/src/constraints.rs (hunk 1 above)
Comment (shamil-gadelshin) [reply runtime-modules/bureaucracy/src/constraints.rs]: Sure!

**Comment by shamil-gadelshin** [comment]
I refactored InputValidationLengthConstraint:
- moved out from the forum to the common module
- updated forum, content_working_group, bureaucracy modules

**Pull request merged**